### ISSUE TITLE
Splunk Operator: BUG - KVstore tries to revert to 3.6.17

### ISSUE THREAD

**Issue by satellite-no** [splunk/splunk-operator#909]
### Please select the type of request

Bug

### Tell us more

**Describe the request**
- Splunk reverts to version 3.6 of mongodb (kvstore) when the standalone splunk pod is recreated after a failure or scaling procedure (ie scale to 0 then back to 1). 

**Expected behavior**
- Splunk should only use version 4.2.17 of mongodb.

**Splunk setup on K8S**
- Details of the Splunk setup on the K8s cluster.

```yaml
---
apiVersion: enterprise.splunk.com/v3
kind: Standalone
metadata:
  name: boris-aio
  namespace: boris
spec:
  image: splunk/splunk:9.0.1
  replicas: 1
  defaults: |-
    splunk:
      http_enableSSL: 1
```

**Reproduction/Testing steps**
- Create new Splunk instance
- Scale down to 0. 
- Scale back up to 1. 

**K8s environment**
- AWS EKS, 1.23.0

**K8s collector data(optional)**
- Please provide data collected from the [K8s collectors](https://github.com/splunk/splunk-operator/blob/develop/docs/K8SCollectors.md), if any.

**Additional context(optional)**
- Add any other context about the problem here.

**Comment by satellite-no** [comment]
Side note this only happens on `image: splunk/splunk:9.0.0` and `image: splunk/splunk:9.0.1` if you set the image too `image: splunk/splunk:8.2.1` the issues does not occur due to KVStore version on that Splunk.

**Comment by sgontla** [comment]
Thanks @satellite-no , i do see the same issue on a pod reset. We will look into this issue.

> [splunk@splunk-shc-search-head-0 splunk]$ grep "db version" mongod.log
 2022-09-27T17:54:11.579Z I  CONTROL  [initandlisten] db version v4.2.17-linux-splunk-v3
 2022-09-27T17:56:45.776Z I  CONTROL  [initandlisten] db version v4.2.17-linux-splunk-v3
 2022-09-27T17:56:52.362Z I  CONTROL  [initandlisten] db version v4.2.17-linux-splunk-v3
 2022-09-27T17:58:56.030Z I  CONTROL  [initandlisten] db version v4.2.17-linux-splunk-v3
 2022-09-27T18:00:00.500Z I  CONTROL  [initandlisten] db version v4.2.17-linux-splunk-v3
 2022-09-27T18:01:09.488Z I  CONTROL  [initandlisten] db version v4.2.17-linux-splunk-v3
 2022-09-27T18:02:20.332Z I  CONTROL  [initandlisten] db version v4.2.17-linux-splunk-v3
 2022-09-27T18:37:34.225Z I CONTROL  [initandlisten] db version v3.6.17-linux-splunk-v4
 2022-09-27T18:38:45.494Z I CONTROL  [initandlisten] db version v3.6.17-linux-splunk-v4
[splunk@splunk-shc-search-head-0 splunk]$ cat /opt/splunk/etc/instance.cfg

**Comment by satellite-no** [comment]
Thanks @sgontla please let me know if there are any workarounds for this issue as you test.  I'd like to deploy a new splunk service at 9.0.1 but this kvstore issue is a critical.

**Comment by satellite-no** [comment]
@sgontla Weird development in this bug that could be an unpleasant work around or help determine root cause. I assume this sets a flag somewhere in `etc/`. 

- Create a net new standalone instance.
- Set the below in `$SPLUNK_HOME/etc/system/local/server.conf`
```text
[kvstore]
storageEngine = mmapv1
```
- Delete the standalone instance and delete the `var/` volume, but let the `etc/` volume persist. 
- Recreate the same instance using the `etc/` persistent volume. 

When the instance comes up it auto upgrades mmapv1 to wiredTiger and presumably sets a flag some where that keeps it from trying to do it again allowing the kvstore to be on v4.2.17 after pod resets or even deletions.

**Comment by sgontla** [comment]
@satellite-no , Thanks for that input. On a Pod reset, the Mongo migration logic is creating a file "/opt/splunk/var/run/splunk/kvstore_upgrade/
versionFile36", which is trying to start the mongod with 3.6.x version, and that is where it is failing. However, if the Pod resets again, it causes the same issue.

**Comment by sgontla** [comment]
@satellite-no , while we are investigating the issue, you may want to use the workaround, once after Pod comes up for the first time:

> 1. touch $SPLUNK_HOME/var/run/splunk/kvstore_upgrade/versionFile42

If the Kvstore is not starting becoz of the Pod reset, then, we can still create this file, and restart splunk, which should act as a workaround.

**Comment by satellite-no** [comment]
Awesome! thats great to know!

**Comment by gavins-k1** [comment]
thanks @sgontla for the workaround 👍

**Comment by pogdin** [comment]
Fixed in 2.1

**Comment by laysbbq** [comment]
Having this issue in 9.2.1. Workaround solved the situation. Thanks!
